Azure/azure-cli
# [App Service] `az webapp up/create/update`: Add new parameter `--basic-auth` to allow users to enable and disable basic auth

**Related command**

az webapp up -g <rg-name> --name <app-name> --basic-auth enabled/disabled
az webapp update -g <rg-name> --name <app-name> --basic-auth enabled/disabled
az webapp create -g <rg-name> -p <serverfarm-name> --name <app-name> --basic-auth enabled/disabled

**Description**

This pr add --basic-auth parameter to webapp create/up/update so that customers can toggle basic auth settings. The default value for --basic-auth is "" (empty string), which will not change existing settings. Customers only allow to input two values: "enabled" and "disabled" case insensitively. Any other values will raise exception. 

**Testing Guide**

create a webapp and use related command to change the basic auth setting
If enabled, you should see basic auth publishing as on. Otherwise off
<img width="495" alt="image" src="https://github.com/Azure/azure-cli/assets/117378021/1327af32-cfb1-421c-876f-c6bb757ae573">


**History Notes**


[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.
